Lesson16/sinem16/script.js

```diff
@@ -357,56 +396,60 @@ console.log("Truthy Values Arr is:", arrWithTruthyValues);
     - Log the final sum.
     - Example: "abc123" -> 6
 */
+console.log("Exp.20");
+
 function sumDigits(str) {
   let sum = 0;
   for (let i = 0; i < str.length; i++) {
-    if (!isNaN(str[i])) {
-      sum += Number(str[i]);
+    let digit = parseInt(str[i]);
+    if (!isNaN(digit)) {
+      sum += digit;
     }
   }
-  return sum;
+  console.log(sum);
 }
-const exampleString = "Code2career130625";
-const sumOfDigits = sumDigits(exampleString);
-console.log(`The sum of the digits in "${exampleString}" is: ${sumOfDigits}`);
+
+sumDigits("11abg11");
 
 /*
 21. Average of Array Elements
     - Define a function `averageArray(numbers)` that uses a loop
       to calculate the average (sum / length).
     - Log the average.
 */
+console.log("Exp.21");
+
 function averageArray(numbers) {
+  let average = 0;
   let sum = 0;
-  for (let i = 0; i < numbers.length; i++) {
+  for (i = 0; i < numbers.length; i++) {
     sum += numbers[i];
   }
-  return sum / numbers.length;
+  average = sum / numbers.length;
+  console.log(average);
 }
-const scores = [75, 80, 95, 60, 90];
-const average = averageArray(scores);
-console.log("The array average is:", average);
+averageArray([1, 2, 3, 4]);
 
 /*
 22. Flatten a 2D Array (Nested Loops)
     - Define a function `flattenArray(twoDArray)` that takes an array of arrays
       (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
     - Log the flattened array.
 */
+console.log("Exp.22");
 
 function flattenArray(twoDArray) {
-  let newArray = [];
+  let newString = [];
   for (let i = 0; i < twoDArray.length; i++) {
-    for (let j = 0; j < twoDArray[i].length; j++) {
-      newArray.push(twoDArray[i][j]);
-    }
+    newString = twoDArray.flat();
```

Comment:
`twoDArray.flat();` is placed inside the loop which is a bug. It runs multiple times and overwrites itself!